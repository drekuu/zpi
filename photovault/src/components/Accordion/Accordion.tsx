'use client';

import RightArrowIcon from '@/../public/icons/right-arrow.svg';
import { ReactNode, useState } from 'react';
import clsx from 'clsx';

interface AccordionProps {
  name: string;
  children: ReactNode;
  childrenClassName?: string;
}

export default function Accordion({
  name,
  children,
  childrenClassName,
}: AccordionProps) {
  const [open, setOpen] = useState(false);

  return (
    <div className='flex flex-col gap-6'>
      <div
        className='flex items-center gap-2 justify-between cursor-pointer'
        onClick={() => setOpen(!open)}
      >
        <p className='text-xl font-bold'>{name}</p>
        <RightArrowIcon
          draggable={false}
          style={{ rotate: open ? '-90deg' : '90deg' }}
        />
      </div>

      <div
        className={clsx(
          childrenClassName,
          open ? 'flex' : 'hidden',
          'overflow-hidden flex flex-col',
        )}
        style={{ height: open ? 'auto' : '0' }}
      >
        {children}
      </div>
    </div>
  );
}
