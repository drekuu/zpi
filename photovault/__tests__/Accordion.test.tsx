import { describe, expect, test } from 'vitest';
import { render, fireEvent } from '@testing-library/react';
import Accordion from '@/components/Accordion/Accordion';

describe('Accordion', () => {
  test('renders name', () => {
    const testName = 'Accordion test';

    const component = render(
      <Accordion name={testName}>
        <div />
      </Accordion>,
    );

    const name = component.getByText(testName);
    expect(name).toBeDefined();
  });

  test('renders children properly [snapshot]', () => {
    const component = render(
      <Accordion name=''>
        <div>
          <p>hello from child</p>
        </div>

        <div>hello from second child</div>
      </Accordion>,
    );

    const children = component.getByTestId('children-test');
    expect(children.children).toMatchSnapshot();
  });

  test('can be opened', () => {
    const testName = 'click test';

    const component = render(
      <Accordion name={testName}>
        <div />
      </Accordion>,
    );

    const name = component.getByText(testName);
    fireEvent.click(name);

    const children = component.getByTestId('children-test');
    expect(children.style.getPropertyValue('height')).not.toBe('0');
  });
});
