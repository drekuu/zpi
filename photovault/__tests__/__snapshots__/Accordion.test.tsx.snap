// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Accordion > renders children properly [snapshot] 1`] = `
HTMLCollection [
  <div>
    <p>
      hello from child
    </p>
  </div>,
  <div>
    hello from second child
  </div>,
]
`;
